### PR TITLE
add basic support for LoongArch

### PR DESCRIPTION
**Pull request by junchao-loongson**
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

Hello friends, recently Loongson released the LoongArch architecture processor, this patch is used to support radare2 can be compiled properly on loongarch.、
I'll be submitting a patch to support binary file analysis for the loongarch platform later.

You can view detailed information about LoongArch at the link below
[LoongArch-Documentation](https://github.com/loongson/LoongArch-Documentation)

### CODE REVIEW

File: libr/include/r_util/r_sys.h [hunk 1]
```diff
@@ -121,6 +121,8 @@ R_API bool r_sys_tts(const char *txt, bool bg);
 #  endif
 #elif __mips__
 #  define r_sys_breakpoint() __asm__ volatile ("break");
+#elif __loongarch__
```
Comment (trufae) [inline libr/include/r_util/r_sys.h:124]: i would prefer to use #elif (__mips__ || __loongarch__)

**Review by trufae — CHANGES_REQUESTED** [review]
thanks! as long as loongson is mips i would prefer to group it as much as possible to the mips logic. are the new toolchains not handling them as mips

**Comment by junchao-loongson** [comment]
> thanks! as long as loongson is mips i would prefer to group it as much as possible to the mips logic. are the new toolchains not handling them as mips

loongarch is not mips, loongarch has new ABI and new Instruction set.
Therefore their toolchain cannot be shared either

**Comment by trufae** [comment]
Got it! do you have any link to learn more about this? Thank you!

**Comment by junchao-loongson** [comment]
> Got it! do you have any link to learn more about this? Thank you!

Of course, here is the English manual of loongarch
https://loongson.github.io/LoongArch-Documentation/README-EN.html

**Pull request merged**